docs/proposals/edge-node-upgrade.md

```diff
@@ -0,0 +1,273 @@
+  filter out nodes that don't meet upgrade requirements(1. edge node already on the desired version. 2. not edge node,
+  without label {"node-role.kubernetes.io/edge": ""} 3. remove duplicated nodes)
+- Upgrade Controller Downstream send Upgrade message to each edge node. And new a Goroutine to process timeout
+  if we don't receive edge node upgrade response, if not, downstream will construct a timeout response message and send it to
+  Upstream to let Upstream update Upgrade status. And patch update Upgrade status state to Upgrading state.
+  And mark edge node unschedulable. If the ReUpgrade flag is true, we will patch update it from true to false, to avoid always reupgrading.
+- CloudHub send upgrade request to EdgeHub
+- EdgeHub receive upgrade message, and upgrade sub module will process upgrade message.
+- EdgeHub Upgrade module validates the upgrade message(checks UpgradeID is null, edgecore is already on the version)
+- To imporve adaptability, we will provide a upgrade provider interface. By default, we will use `keadm` to do upgrade
+  operation. Users can set `UpgradeTool` and `UpgradeType` field to choose other installers to finish upgrade.
+- KeadmUpgrade will download the keadm of the corresponding version. EdgeCore will pull kubeedge/installation-package
+  image, and copy the keadm binary from container to hostpath. And then run `keadm upgrade` related command to do upgrade operation.
+
+- keadm upgrade EdgeCore and report upgrade result to CloudHub Http service.
+
+- keadm use `/etc/kubeedge/idempotency_record` file to district only one upgrade can be performed at a time.
+
+- keadm reprocess: back up `edgecore.db edgecore.yaml edgecore` to backup path `/etc/kubeedge/backup/{From_Version}`,
+  and pull `kubeedge/installation-package` image and copy the edgecore binary from container to host upgrade path:
+  `/etc/kubeedge/upgrade/{To_Version}`
+
+- keadm process: stop the edgecore, and copy the new version edgecore to `/usr/local/bin` directory and start new edgecore
+
+- keadm rollback: stop the edgecore, rollback files, copy files of backup dir `/etc/kubeedge/backup/{From_Version}` to
+  the origin path. and start origin edgecore.
+
+- keadm will report upgrade result to CloudHub Http service in the defer function.
+
+- CloudHub Http service will add an interface `/upgrade` to transform the upgrde response message to the Upgrade
+  Controller Upstream.
+
+- Upgrade Controller Upstream mark node schedulable, and patch upgrade status.
+
+
+<img src="../images/edge-node-upgrade/upgrade.png">
+
+
+## CRD Design Details
+
+### CRD API Group and Version
+The `Upgrade` CRD will be cluster-scoped.
+The tables below summarize the group, kind and API version details for the CRDs.
+
+* Upgrade
+
+| Field                 | Description             |
+|-----------------------|-------------------------|
+|Group                  | upgrade.kubeedge.io     |
+|APIVersion             | v1alpha2                |
```

Comment:
why the init version is not v1alpha1?

Reply:
already fixed to v1alpha1 in source codes, will also update the proposal